MovieDB/MovieDB/Application/Scenes/Favorites/FavoritesViewModel.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  FavoritesViewModel.swift
+//  MovieDB
+//
+//  Created by pham.van.ducd on 5/22/19.
+//  Copyright © 2019 tranductanb. All rights reserved.
+//
+
+struct FavoritesViewModel {
+    let navigator: FavoritesNavigatorType
+    let useCase: FavoritesUseCaseType
+}
+
+// MARK: - ViewModelType
+extension FavoritesViewModel: ViewModelType {
+    struct Input {
+        let loadTrigger: Driver<Void>
+        let reloadTrigger: Driver<Void>
+        let loadMoreTrigger: Driver<Void>
+        let selectFavoritesTrigger: Driver<IndexPath>
+    }
+    
+    struct Output {
+        let error: Driver<Error>
+        let loading: Driver<Bool>
+        let refreshing: Driver<Bool>
+        let loadingMore: Driver<Bool>
+        let fetchItems: Driver<Void>
+        let favoritesList: Driver<[Favorite]>
+        let selectedFavorites: Driver<Void>
+        let isEmptyData: Driver<Bool>
+    }
+    
+    func transform(_ input: Input) -> Output {
+        let loadMoreOutput = setupLoadMorePaging(
+            loadTrigger: input.loadTrigger,
+            getItems: useCase.getFavoriteList,
+            refreshTrigger: input.reloadTrigger,
+            refreshItems: useCase.getFavoriteList,
+            loadMoreTrigger: input.loadMoreTrigger,
+            loadMoreItems: useCase.loadMoreFavoriteList)
+        
+        let (page, fetchItems, loadError, loading, refreshing, loadingMore) = loadMoreOutput
+        
+        let favoritesList = page
+            .map { $0.items.map { $0 } }
+            .asDriverOnErrorJustComplete()
+        
+        let selectedFavorites = input.selectFavoritesTrigger
+            .withLatestFrom(favoritesList) {($0, $1)}
```

Comment:
```suggestion
            .withLatestFrom(favoritesList) { indexPath, favorites in
                 return favorites[indexPath.row]
            }
```
cho luôn như này ko cần dùng map bên dưới anh nhé